app/models/user.rb

```diff
@@ -47,4 +47,20 @@ class User < ApplicationRecord
   after_update if: [:verified_previously_changed?, :verified?] do
     events.create! action: "email_verified"
   end
+
+  def self.from_omniauth(access_token)
+    provider_uid = access_token.uid
+    email = access_token.info['email']
+
+    user = User.find_or_create_by(email:, provider_uid:)
+    set_secure_password(user) unless user.password_digest.present?
```

Comment:
You can [pass a block](https://api.rubyonrails.org/classes/ActiveRecord/Relation.html#method-i-find_or_create_by) for the `_create_` version.

```suggestion
    User.find_or_create_by(email:, provider_uid:) do |user|
      user.password = SecureRandom.hex(10)
    end
```